rust-symcrypt/src/rsa/mod.rs

```diff
@@ -0,0 +1,755 @@
+                    inner: rsa_key,
+                    rsa_key_usage: rsa_key_usage,
+                    has_private_key: true,
+                }),
+                err => Err(err.into()),
+            }
+        }
+    }
+
+    /// `set_key_pair()` sets both the public and private key information onto [`RsaKey`].
+    ///
+    /// `modulus_buffer` takes in a `&[u8]` reference to a byte array that contains the modulus of the Rsa key.
+    ///
+    /// `pub_exp` takes in a `&[u8]` that is the public exponent represented by an array of bytes.
+    ///
+    /// `p` takes in a `&[u8]` reference to a byte array that contains the first prime.
+    ///
+    /// `q` takes in a `&[u8]` reference to a byte array that contains the second prime.
+    pub fn set_key_pair(
+        modulus_buffer: &[u8],
+        pub_exp: &[u8],
+        p: &[u8],
+        q: &[u8],
+        rsa_key_usage: RsaKeyUsage,
+    ) -> Result<Self, SymCryptError> {
+        let n_bits_mod = (modulus_buffer.len() as symcrypt_sys::SIZE_T) * 8; // Convert the size from bytes to bits.
+        let rsa_key = allocate_rsa(2, n_bits_mod)?;
+        let u64_pub_exp = load_msb_first_u64(pub_exp)?;
+
+        // Construct the primes_ptr and primes_len_ptr for SymCryptRsaKeyValue consumption
+        let primes_ptr = [p.as_ptr(), q.as_ptr()].as_mut_ptr();
+        let primes_len_ptr = [
+            p.len() as symcrypt_sys::SIZE_T,
+            q.len() as symcrypt_sys::SIZE_T,
+        ]
+        .as_mut_ptr();
+        unsafe {
+            // SAFETY: FFI calls
+            match symcrypt_sys::SymCryptRsakeySetValue(
+                modulus_buffer.as_ptr(),
+                modulus_buffer.len() as symcrypt_sys::SIZE_T,
+                [u64_pub_exp].as_ptr(),    // This array has a length of 1.
+                1 as symcrypt_sys::UINT32, // Must be 1.
+                primes_ptr,
+                primes_len_ptr,
+                2 as symcrypt_sys::UINT32,
+                NumberFormat::MSB.to_symcrypt_format(),
+                rsa_key_usage.to_symcrypt_flag(),
+                rsa_key.0,
+            ) {
+                symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => Ok(RsaKey {
+                    inner: rsa_key,
+                    rsa_key_usage: rsa_key_usage,
+                    has_private_key: true,
+                }),
+                err => Err(err.into()),
+            }
+        }
+    }
+
+    /// `set_public_key()` sets only the public key information onto the [`RsaKey`].
+    ///
+    /// `modulus_buffer` takes in a `&[u8]` reference to a byte array that contains the modulus of the Rsa key.
+    ///
+    /// `pub_exp` takes in a `&[u8]` that is an array of bytes representing the public exponent.
+    ///
+    /// `rsa_key_usage` takes in a [`RsaKeyUsage`] and will indicate if this key will be used for [`RsaKeyUsage::Sign`], or [`RsaKeyUsage::Encrypt`], or [`RsaKeyUsage::SignAndEncrypt`]
+    pub fn set_public_key(
```

Comment:
>  pub fn set_public_key( [](http://example.com/codeflow?start=2&length=24)

We definitely don't want to exit when importing a bogus RSA key coming from a untrusted source.